presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -567,7 +569,9 @@ private List<DataOutput> bufferStripeData(long stripeStartOffset, FlushReason fl
                     .mapToLong(StreamDataOutput::size)
                     .sum();
         }
-        streamLayout.reorder(dataStreams, ImmutableMap.of(), ImmutableMap.of());
+        Map<Integer, ColumnEncoding> columnEncodings = new HashMap<>();
```

Comment:
1. Let's do an immutable map instead since we are going to pass it around.
2. Before building the columns encodings, add 0 node, it's line 598 in the original source. 
```
// the 0th column is a struct column for the whole row
        columnEncodings.put(0, new ColumnEncoding(DIRECT, 0));
```